src/lib.rs

```diff
@@ -303,3 +191,167 @@ fn generate_analysis(config: &Config) -> Result<(), Box<std::error::Error>> {
+                DefKind::Type => (),
+                DefKind::Local => (),
+                DefKind::Field => (),
+            }
+        }
+
+        Ok(data)
+    }
+
+    fn to_json(&self, config: &Config) -> Result<String, Box<std::error::Error>> {
+        use jsonapi::api::*;
+
+        let root_def = config.host.get_def(self.krate.id)?;
+
+        let mut document = JsonApiDocument::default();
+
+        let mut map = HashMap::new();
+        map.insert(
+            String::from("docs"),
+            serde_json::Value::String(root_def.docs.clone()),
+        );
+
+        let mut relationships = HashMap::new();
+
+        let mut relationship = Relationship {
+            data: IdentifierData::Multiple(Vec::new()),
+            links: None,
+        };
+
+        //TODO this is bad, use real option handling in the loop
+        document.included = Some(Vec::new());
+
+        for (id, item) in self.data.iter() {
+            match item {
+                &Item::Module { ref name, ref docs } => {
+                    if let IdentifierData::Multiple(ref mut v) = relationship.data {
+                        v.push(ResourceIdentifier {
+                            _type: String::from("module"),
+                            id: id.clone(),
+                        });
+                    };
+                    let mut map = HashMap::new();
+                    map.insert(
+                        String::from("name"),
+                        serde_json::Value::String(name.clone()),
+                    );
+                    map.insert(
+                        String::from("docs"),
+                        serde_json::Value::String(docs.clone()),
+                    );
+
+                    let module = Resource {
+                        _type: String::from("module"),
+                        id: id.clone(),
+                        attributes: map,
+                        links: None,
+                        meta: None,
+                        relationships: None,
+                    };
+
+                    document.included.as_mut().map(|v| v.push(module));
+                }
+            }
+        }
+
+        relationships.insert(String::from("modules"), relationship);
+
+        let len = root_def.qualname.len();
+        let krate = Resource {
+            _type: String::from("crate"),
+            // example:: -> example
+            id: root_def.qualname[..(len - 2)].to_string(),
+            attributes: map,
+            links: None,
+            meta: None,
+            relationships: Some(relationships),
+        };
+
+        document.data = Some(PrimaryData::Single(Box::new(krate)));
+
+        serde_json::to_string(&document).map_err(|e| Box::new(e) as Box<std::error::Error>)
+    }
+}
+
+#[derive(Debug)]
+struct Crate {
+    id: analysis::Id,
+    name: String,
+    docs: String,
+    modules: Vec<String>,
+}
+
+#[derive(Debug)]
+enum Item {
+    Module { name: String, docs: String },
```

Comment:
This will be expanded over time I'm guessing based off the `DefKind` matching block above right?